File: LinkedList/DoubleNode.py
# -*- coding: utf-8 -*-
"""
Created on Tue Jun  2 16:51:30 2020

@author: Kakatuahitam
"""


class DoubleNode:
    def __init__(self, prev = None, data = None, next = None):
        self.prev = prev
        self.data = data
        self.next = next
        
    def __repr__(self):
        return repr(self.data)

File: LinkedList/DoubleNode.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package linkedlist;

/**
 *
 * @author Kakatuahitam
 * source: https://dzone.com/articles/doubly-linked-list-in-java
 */
class DoubleNode<Type> {
    Type data;
    DoubleNode<Type> next;
    DoubleNode<Type> prev;
    
    DoubleNode(Type data){
        this(null, data, null);
    }
    
    DoubleNode(DoubleNode<Type> prev, Type data, DoubleNode<Type> next){
        this.data = data;
        this.next = next;
        this.prev = prev;
    }
}
